app/models/manageiq/providers/nuage/network_manager/cloud_subnet.rb

```diff
@@ -1,2 +1,60 @@
 class ManageIQ::Providers::Nuage::NetworkManager::CloudSubnet < ::CloudSubnet
+  supports :delete
+  supports :create
+
+  def delete_cloud_subnet_queue(userid)
+    task_opts = {
+      :action => "deleting Cloud Subnet for user #{userid}",
+      :userid => userid
+    }
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'raw_delete_cloud_subnet',
+      :instance_id => id,
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => ext_management_system.my_zone,
+      :args        => []
+    }
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def raw_delete_cloud_subnet
+    $nuage_log.info("Deleting Cloud Subnet (ems_ref = #{ems_ref})")
+
+    Ansible::Runner.run(
+      ext_management_system.ansible_env_vars,
+      ext_management_system.ansible_extra_vars(:id => ems_ref),
+      ext_management_system.playbook('remove-subnet.yml')
+    )
```

Comment:
I like how this came out

Reply:
AnsibleRunner is love at first sight for me as well 😍